### ISSUE TITLE
Cannot launch internal binary editor

### ISSUE THREAD

**Issue by Andreas0602** [mucommander/mucommander#938]
I've read that muCommander has an internal binary editor, but I just can't manage to launch it.

In many cases, when I select a binary file (for example a PDF file) and press F4, I get the following error:

<img width="360" alt="Screenshot_2023-05-17_20-41-25" src="https://github.com/mucommander/mucommander/assets/79788982/eb9c0433-35cc-4c59-a8fc-395e1ae03614">

In other cases, muCommander opens the file in its internal editor in text mode, but when I then go to the menu item Editor > Binary, nothing happens.

By the way, the internal binary **viewer** does work fine.

Here's my OS version and muCommander version:

<img width="570" alt="Screenshot_2023-05-17_20-46-33" src="https://github.com/mucommander/mucommander/assets/79788982/74f0a82c-b11e-4085-a204-66cdd16703fb">

**Comment by hajdam** [comment]
Hi, thanks for reporting. I personally don't have a Mac, so I cannot reproduce this issue.

First error could be related to issue #983 - if you could try if the existing fix affects it.

For second issue, could you check if there is any exception reported? In Help/Debug console set Level to INFO or higher and then try binary editor and check back to debug console if there is any new errors reported.

**Comment by pskowronek** [comment]
> First error could be related to issue #983 - if you could try if the existing fix affects it.

it only affected Text Editor/Viewer, not Binary one.

**Comment by hajdam** [comment]
> > First error could be related to issue #983 - if you could try if the existing fix affects it.
> 
> it only affected Text Editor/Viewer, not Binary one.

As far as I'm aware, pressing F4 always opens internal text editor first, only after that you can switch to binary editor.

**Comment by pskowronek** [comment]
if it is the case, then it could have been the potential reason.

**Comment by pskowronek** [comment]
@Andreas0602 can you provide a sample file where you reproduce the problem?

**Comment by pskowronek** [comment]
On released version (v1.2.0) the provided file opens in Text Editor (and, that's strange, on second retry in Binary Editor).

On the master version I kinda reproduce the problem and here is stacktrace (@hajdam):
```
2023-07-02 00:42:35.899 [Thread-11] ERROR com.mucommander.ui.viewer.FileFrame:79 - Exception caught
java.lang.NullPointerException: null
        at com.mucommander.viewer.binary.BinaryEditor.updateUndoStatus(BinaryEditor.java:239)
        at com.mucommander.viewer.binary.BinaryEditor.init(BinaryEditor.java:210)
        at com.mucommander.viewer.binary.BinaryEditor.<init>(BinaryEditor.java:101)
        at com.mucommander.viewer.binary.BinaryFileViewerService.createFileEditor(BinaryFileViewerService.java:71)
        at com.mucommander.ui.viewer.FileEditorPresenter.switchFileEditor(FileEditorPresenter.java:215)
        at com.mucommander.ui.viewer.FileEditorPresenter.show(FileEditorPresenter.java:185)
        at com.mucommander.ui.viewer.FilePresenter.open(FilePresenter.java:161)
        at com.mucommander.ui.viewer.FileFrame$1.getTargetComponent(FileFrame.java:77)
        at com.mucommander.ui.layout.AsyncPanel$2.run(AsyncPanel.java:109)
```
